### DISCUSSION TITLE
FFMPEG issue - Video preview not working

### DISCUSSION THREAD

**Discussion by ghost** [pimcore/pimcore#10581] [Q&A / Support] [answered]
Hi,  I have  installed v.10.1.4 on a dev server. ffmpeg is installed and the assets are stored on local db. I can preview images and pdf but the video preview does not work even if you add ffmpeg to PATH

thank you very much

![image](https://user-images.githubusercontent.com/68632142/137107248-5d2acad5-0613-44f7-bd04-098e55fa1270.png)  

---
var/log

![image](https://user-images.githubusercontent.com/68632142/137111143-aaf7e6dc-cebe-4571-9d9d-f67a528eedc9.png)

dev.log =>  pimcore.ERROR: Exception: No 'ffmpeg' executable found, please install the application or add it to the PATH (in system settings or to your PATH environment variable in

![image](https://user-images.githubusercontent.com/68632142/137111253-eab0819a-76b7-48c9-9d19-49cdfb4cccb6.png)

ffmpeg version
![image](https://user-images.githubusercontent.com/68632142/137107120-15a670b3-8578-490b-ba14-a31e3d2148c1.png)

var/tmp

![image](https://user-images.githubusercontent.com/68632142/137107912-5d3c24eb-f56c-40aa-a082-2d65e24b98e9.png)

**Comment by ghost** [comment, marked as answer]
Fixed:

The videopreview started as soon as I manually customized the full path to the ffmpeg executable on the server.
The preview works both with local storage and external storage S3

simply replaced entry in config/services.yaml
before:  #pimcore_executable_ffmpeg: /opt/tools/ffmpeg
after:       pimcore_executable_ffmpeg: /usr/bin/ffmpeg  

and cleared cache with bin/console cache:clear

see:
https://pimcore.com/docs/pimcore/current/Development_Documentation/Installation_and_Upgrade/System_Setup_and_Hosting/Additional_Tools_Installation.html

https://github.com/pimcore/skeleton/blob/10.x/config/services.yaml

**Comment by grafpaper1987** [comment]
I'm experiencing the same issue. Updated the ffmpeg executable path and still am getting stuck on "The preview for this video is currently in progress. Please wait ..."

![image](https://github.com/pimcore/pimcore/assets/28054885/f64eb093-34ea-4835-88b8-ccc66bc49546)

![image](https://github.com/pimcore/pimcore/assets/28054885/22826933-9be5-4df0-afe3-a2da0f933c8c)

**Comment by ghost** [comment]
Hi, in the root folder of your pimcore installation you should find the services.yaml file in the config folder
Form me its like that

 

***@***.*** pimcore]$ ls -l

total 704

drwxrwxr-x  2 delph delph   4096 May 15 16:50 bin

-rw-rw-r--  1 delph delph   1796 Nov 14  2022 composer.json

-rw-rw-r--  1 delph delph 640321 Nov 14  2022 composer.lock

drwxr-xr-x  6 delph delph    152 Jul  4  2022 config

-rw-rw-r--  1 delph delph   1136 Nov  3  2022 docker-compose.yml

-rw-r--r--  1 delph delph     96 Apr  5 14:44 error_log

-rw-rw-r--  1 delph delph  35147 Nov  7  2022 gpl-3.0.txt

drwxr-x---  4 delph nobody    12288 May 15 16:50 public

drwxr-xr-x  8 delph delph    123 Nov  4  2022 src

drwxr-xr-x  3 delph delph     20 May 28  2021 templates

drwxr-xr-x  2 delph delph     21 May 28  2021 translations

drwxr-xr-x 13 delph delph   4096 Dec 20 15:19 var

drwxrwxr-x 85 delph delph   4096 Nov 14  2022 vendor






***@***.*** pimcore]$ ls config

bundles.php  config.yaml  local  packages  pimcore  preload.php  routes  routes.yaml  services.yaml

**Comment by ghost** [comment]
Checkk the correct path of your ffmpeg execuble  

***@***.*** pimcore]$ which ffmpeg

/usr/bin/ffmpeg


then in config/services.yaml set the path directly:

pimcore_executable_ffmpeg: /usr/bin/ffmpeg

**Comment by grafpaper1987** [comment]
Thank you! See below. I think I've got the services.yaml file configured correctly but my video thumbnails are still hanging.

![image](https://github.com/pimcore/pimcore/assets/28054885/e6aa5ab7-bf6a-4691-91a8-c1519a67acac)


![image](https://github.com/pimcore/pimcore/assets/28054885/fb7e14a9-eb3b-43e7-a387-4b625727a3a2)

**Comment by ghost** [comment]
Well, if you use external storage like AWS S3 then it takes some time to download the assets to local disk and then provide the preview

**Comment by ghost** [comment]
You must take care of the indentation in your yaml file !
